src/security/kuzzleSecurity.js

```diff
@@ -709,5 +709,115 @@ KuzzleSecurity.prototype.userFactory = function(id, content) {
+ * @param {string} collection - (optional) The name of the collection to perform the action onto.
+ *
+ * @returns {string} ['allowed', 'denied', 'conditional'] where conditional cases
+ *                   correspond to rights containing closures.
+ *                   See also http://kuzzle.io/guide/#roles-definition
+ */
+KuzzleSecurity.prototype.isActionAllowed = function(policies, controller, action, index, collection) {
+  var filteredPolicies;
+
+  if (!policies || typeof policies !== 'object') {
+    throw new Error('policies parameter is mandatory for isActionAllowed function');
+  }
+  if (!controller || typeof controller !== 'string') {
+    throw new Error('controller parameter is mandatory for isActionAllowed function');
+  }
+  if (!action || typeof action !== 'string') {
+    throw new Error('action parameter is mandatory for isActionAllowed function');
+  }
+
+  // We filter in all the policies that match the request (including wildcards).
+  filteredPolicies = policies.filter(function (policy) {
+    return policy.controller === controller || policy.controller === '*';
+  })
+  .filter(function (policy) {
+    return policy.action === action || policy.action === '*';
+  })
+  .filter(function (policy) {
+    return policy.index === index || policy.index === '*';
+  })
+  .filter(function (policy) {
+    return policy.collection === collection || policy.collection === '*';
+  });
+
+  // Then, if at least one policy allows the action, we return 'allowed'
+  if (filteredPolicies.some(function (item) { return item.value === 'allowed'; })) {
+    return 'allowed';
+  }
+  // If no policy allows the action, we check for conditionals.
+  if (filteredPolicies.some(function (item) { return item.value === 'conditional'; })) {
+    return 'conditional';
+  }
+  // Otherwise we return 'denied'.
+  return 'denied';
+};
+
+
+/**
+ * Gets the rights array of a given user.
+ *
+ * @param  {string} userId The id of the user.
+ * @param  {function} cb   The callback containing the normalized array of rights.
+ */
+KuzzleSecurity.prototype.getUserRights = function (userId, options, cb) {
+  var data = {_id: userId};
+
+  if (!userId || typeof userId !== 'string') {
+    throw new Error('userId parameter is mandatory for isActionAllowed function');
+  }
+
+  if (!cb && typeof options === 'function') {
+    cb = options;
+    options = null;
+  }
+
+  if (cb) {
+    this.kuzzle.query(this.buildQueryArgs('getUserRights'), data, options, function (err, res) {
+      if (err) {
+        return cb(err);
+      }
+
+      cb(null, res.result.hits);
+    });
+  } else {
+    this.kuzzle.query(this.buildQueryArgs('getUserRights'), data, options);
+  }
+};
+
+/**
+ * Gets the rights array of the currently logged user.
+ *
+ * @param  {function} cb The callback containing the normalized array of rights.
+ */
+KuzzleSecurity.prototype.getMyRights = function (options, cb) {
+  if (!cb && typeof options === 'function') {
```

Comment:
Same here, callback should be mandatory, otherwise there is no way to get the result.